src/openrct2/object/AudioSampleTable.h

```diff
@@ -0,0 +1,54 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2022 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../audio/AudioSource.h"
+#include "../core/JsonFwd.hpp"
+
+struct IReadObjectContext;
+
+class AudioSampleTable
+{
+private:
+    struct Entry
+    {
+        OpenRCT2::Audio::IAudioSource* Source{};
```

Comment:
Why isn't this a smart pointer.

Reply:
Sources are stored as smart pointers, but on the mixer.